### PR TITLE
Modify lastSyncTime logic.

### PR DESCRIPTION
**Pull request by amwat**
The Gerrit REST API does not always list up-to-date changes.

i.e. due to unknown delays the following scenario can occur chronologically:
an actual gerrit change -> our sync loop -> change is visible in the API

This makes us miss events that are "just before" our sync loop.
The easiest bet is to mark our sync time to the latest updated change time that the gerrit api shows so that we will catch these events in the next sync loop at the least.

/cc @krzyzacy

### CODE REVIEW

File: prow/gerrit/adapter/adapter.go [hunk 1]
```diff
@@ -150,13 +150,16 @@ func (c *Controller) SaveLastSync(lastSync time.Time) error {
 // and creates prowjobs according to specs
 func (c *Controller) Sync() error {
 	// gerrit timestamp only has second precision
```
Comment (krzyzacy) [inline prow/gerrit/adapter/adapter.go]: nit: I think you can drop this comment

Comment (amwat) [reply prow/gerrit/adapter/adapter.go]: done

**Pull request merged**